### PR TITLE
Frontend integration

### PR DESCRIPTION
**Pull request by Bhatt21**

### CODE REVIEW

File: routes/o2auth_google.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
+const express = require('express');
+const { OAuth2Client } = require('google-auth-library');
+const router = express.Router();
+const User = require('../models/user'); 
+const jwt = require('jsonwebtoken');
+
+// Initialize Google OAuth client
+const googleClient = new OAuth2Client('243962731858-rb04b0dbelevp5blhc339io995opuhqu.apps.googleusercontent.com');
```
Comment (juyee1698) [inline routes/o2auth_google.js]: Are you sure we can add this in public repository? Can you create an environment variable for the key?

File: controllers/auth.js [hunk 2]
```diff
@@ -106,7 +106,7 @@ exports.login = (req, res, next) => {
         );
         res.status(200).json({
             token: token,
-            userId: loadedUser._id.toString()
+            userId: loadedUser.name.toString()
```
Comment (juyee1698) [inline controllers/auth.js]: Can you change this to ID again? We need User ID to get user information and track user's activity in the session.

File: routes/o2auth_google.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+                email,
+                password: "xxx"
+            });
+        }
+
+        // Generate JWT token
+        const token = jwt.sign({ userId: user._id }, "somesuperprojectsecret", { expiresIn: '1h' });
+
+        // Send JWT token and user ID to the frontend
+        res.status(201).json({ token, userId: name });
```
Comment (juyee1698) [inline routes/o2auth_google.js]: Can you change it from name to user ID here as well?

File: routes/o2auth_google.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        const { sub, email, name, picture } = payload; // Extract user information
+        console.log(email, name)
+        let user = await User.findOne({ email });
+
+        if (!user) {
+            // User doesn't exist, create a new user
+            user = await User.create({
+                name,
+                email,
+                password: "xxx"
```
Comment (juyee1698) [inline routes/o2auth_google.js]: A couple of changes here. Please encrypt the password. And we also need to add the following attributes in the user model:
imageUrl:'images/user/profile.jpg',
city:'',
country:'',
address:'',
phoneno:0,
postal:0

**Review by juyee1698 — CHANGES_REQUESTED** [review]

File: controllers/auth.js (hunk 2 above)
Comment (Bhatt21) [reply controllers/auth.js]: ah I see, i will re add that, but we also need to send name, that's what we will show, something like Hi Juyee on the top right when u login

**Review by juyee1698 — APPROVED** [review]

**Comment by juyee1698** [comment]
LGTM

**Pull request merged**